cs3api4lab/api/locks/local_lock_api.py

```diff
@@ -0,0 +1,69 @@
+import json
+import time
+import datetime
+import urllib.parse
+
+from cs3api4lab.api.cs3_user_api import Cs3UserApi
+from cs3api4lab.api.locks.lock_base import LockBase
+
+
+class LocalLockApi(LockBase):
+
+    user = None
+
+    def __init__(self, log, config):
+        super().__init__(log, config)
+        self.log = log
+        self.config = config
+        self.user_api = Cs3UserApi(log)
+        self.locks_expiration_time = self.config.locks_expiration_time
+
+    def handle_locks(self, file_path, endpoint):
+        lock = self._get_lock(file_path, endpoint)
+        if not lock:
+            self._lock_file(file_path, endpoint)
+            return
+        else:
+            if self._is_lock_mine(lock):
+                self._lock_file(file_path, endpoint)
+                return
+            if self._is_lock_expired(lock):
+                self._lock_file(file_path, endpoint)
+                return
+        raise IOError("File locked")
+
+    def _generate_lock_entry(self):
+        user = self.get_current_user()
+        return {self.get_my_lock_name(): urllib.parse.quote(json.dumps({
+            "username": user.username,
+            "idp": user.id.idp,
+            "opaque_id": user.id.opaque_id,
+            "updated": time.time(),
+            "created": time.time()
+        }))}
+
+    def _lock_file(self, file_path, endpoint):
+        self.storage_api.set_metadata(self._generate_lock_entry(), file_path, endpoint)
+
+    def is_valid_external_lock(self, file_path, endpoint):
+        lock = self._get_lock(file_path, endpoint)
+        is_mine = self._is_lock_mine(lock)
+        return lock and not is_mine and not self._is_lock_expired(lock)
```

Comment:
this looks like a bug, i think this should be something like this: if the lock is not mine OR if it's not expired